src/agent/src/device.rs

```diff
@@ -408,6 +416,39 @@ async fn get_vfio_pci_device_name(
     Ok(format!("{}/{}", SYSTEM_DEV_PATH, &uev.devname))
 }
 
+#[cfg(target_arch = "s390x")]
+#[derive(Debug)]
+struct VfioApMatcher {
+    syspath: String,
+}
+
+#[cfg(target_arch = "s390x")]
+impl VfioApMatcher {
+    fn new(device: ap::Device) -> VfioApMatcher {
+        VfioApMatcher {
+            syspath: format!(
+                "{}/card{:02x}/{}",
+                AP_ROOT_BUS_PATH, device.adapter_id, device
+            ),
+        }
+    }
+}
+
+#[cfg(target_arch = "s390x")]
+impl UeventMatcher for VfioApMatcher {
+    fn is_match(&self, uev: &Uevent) -> bool {
+        uev.action == "add" && uev.devpath == self.syspath
```

Comment:
I do wonder if we should make a common matcher which just checks syspath, for all the cases where we can determine the exact sysfs path ahead of time.  Then we can just have multiple constructors or helper functions to generate that matcher based on whatever info we have.

That can be a clean up for another time though.